Persist trading pair in the Cfd

klochowicz opened the pull request:
Thankfully, TradingPair was part of the Offer, so this change should be fully backwards-compatible.

- remove hardcoded BtcUsd in projection,
- persist trading_pair received from the Offer in all Cfd tables,
- default to TradingPair::BtcUsd in already created CFDs.

Fixes #2458

bonomat commented on sqlite-db/migrations/20220728115915_add_trading_pair.sql:
```diff
@@ -0,0 +1,16 @@
+ALTER TABLE
+    cfds
+ADD
+    COLUMN trading_pair NOT NULL DEFAULT 'BtcUsd';
```
We are starting to introduce different terminologies 😬 
In the metrics I've used the term `symbol`. 

I've created an ADR for this so that we are all on the same page: https://github.com/itchysats/architecture/pull/35/files

bonomat replied:
I'm aware that I was the one who introduced `trading symbol` 🙈

klochowicz replied:
all good, I was just following what was written in the ticket.... will update when we decide on `symbol` vs `contract_symbol`.

da-kami commented on sqlite-db/src/lib.rs:
```diff
@@ -185,8 +187,9 @@ impl Connection {
-            initial_tx_fee_rate
-        ) values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13)"#,
+            initial_tx_fee_rate,
+            trading_pair,
```
Here is where CI currently fails because of our friend the `,` :)

da-kami commented on sqlite-db/src/closed.rs:
```diff
@@ -129,7 +129,8 @@ impl Connection {
                 lock_txid as "lock_txid: models::Txid",
-                lock_dlc_vout as "lock_dlc_vout: models::Vout"
+                lock_dlc_vout as "lock_dlc_vout: models::Vout",
+                trading_pair as "trading_pair: models::TradingPair"
```
🛠 Something it odd here! we are loading it but not inserting it into `closed_cfds` - same for `failed_cfds`!

klochowicz replied:
added! I noted that 2 other parameters from the Cfd are missing in FailedCfd - was that on purpose???

da-kami reviewed (dismissed):
I think `insert` for `closed_cfds` and `failed_cfds` is missing.

bonomat committed:
Rename `trading_pair` to `contract_symbol`

bonomat committed:
Persist trading pair in the Cfd

Thankfully, TradingPair was part of the Offer, so this change should be fully backwards-compatible.

- remove hardcoded BtcUsd in projection,
- persist trading_pair received from the Offer in all Cfd tables,
- default to TradingPair::BtcUsd in already created CFDs.

Co-authored-by: Mariusz Klochowicz <mariusz@klochowicz.com>

bonomat reviewed (approved):
Approving in the name of @klochowicz

da-kami commented on CHANGELOG.md:
```diff
@@ -16,6 +16,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Replace intermediate confirmation step in rollover protocol for the maker with a configurable flag which can be updated during runtime using the new `POST /rollover/config` endpoint.
   On startup, rollovers are configured to be accepted by default.
+- Breaking change: renamed `trading_pair` to `contract_symbol` in Http API.
```
```suggestion
- Renamed `trading_pair` to `contract_symbol` in Http API.
```

🙃 At some point we should revisit our discussion about what a "breaking change" is. I would not state it like this in the changelog.
For us internally this is a breaking change, but for users it should not matter because the network protocols are not affected. 
If anybody looks at the changelog this looks like this version is incompatible to the one before, but that is misleading.

luckysori replied:
Hmm, you are assuming that no one is trying to run their own `maker` automation then. And that this changelog is only meant to be consumed by users of the `taker` binary.

But I agree that our compatibility story is quite confusing.

klochowicz replied:
AFAICT; the PUT request didn't have `trading_pair` - it was only for the UI

bonomat replied:
For me, the changelog is for developers but  yea, eventually we should revisit this discussion :)

da-kami commented on model/src/cfd.rs:
```diff
@@ -196,7 +196,7 @@ impl From<Origin> for Role {
     pub id: OfferId,
 
-    pub trading_pair: TradingPair,
+    pub contract_symbol: ContractSymbol,
```
🛠 I think this is not only a breaking change in the HTTP API but a breaking change on the network layer! I think we need a `#[serde(rename = "trading_pair")]` here, otherwise already rolled out `0.5.x` won't be compatible.

da-kami replied:
Even though this field is not used, I think we would still run into trouble deserializing the `Offer` if the maker sends it out like this, because the field will be missing on the taker side.

klochowicz replied:
good point. We need to use serde rename here indeed. I'll add a fixup.
